Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+        var recipientPublicKey = secp256k1_pubkey()
+        guard secp256k1_ec_pubkey_parse(secp256k1.Context.rawRepresentation, &recipientPublicKey, bytes, bytes.count) != 0 else {
+            throw DirectMessageEncryptingError.pubkeyParsing
+        }
+        return recipientPublicKey
+    }
+
+    private func computeSharedSecret(using publicKey: secp256k1_pubkey, and privateKeyBytes: [UInt8]) throws -> [UInt8] {
+        var sharedSecret = [UInt8](repeating: 0, count: 32)
+        var mutablePublicKey = publicKey
+        guard secp256k1_ecdh(secp256k1.Context.rawRepresentation, &sharedSecret, &mutablePublicKey, privateKeyBytes, { (output, x32, _, _) in
+            memcpy(output, x32, 32)
+            return 1
+        }, nil) != 0 else {
+            throw DirectMessageEncryptingError.sharedSecretCreation
+        }
+        return sharedSecret
+    }
+
+    private func AESDecrypt(data: [UInt8], iv: [UInt8], sharedSecret: [UInt8]) -> Data? {
+        return AESOperation(operation: CCOperation(kCCDecrypt), data: data, iv: iv, sharedSecret: sharedSecret)
+    }
+
+    private func AESEncrypt(data: [UInt8], iv: [UInt8], sharedSecret: [UInt8]) -> Data? {
+        return AESOperation(operation: CCOperation(kCCEncrypt), data: data, iv: iv, sharedSecret: sharedSecret)
+    }
+
+    private func encodeDMBase64(content: [UInt8], iv: [UInt8]) -> String {
+        let contentBase64 = Data(content).base64EncodedString()
+        let ivBase64 = Data(iv).base64EncodedString()
+        return contentBase64 + "?iv=" + ivBase64
+    }
+
+    private func AESOperation(operation: CCOperation, data: [UInt8], iv: [UInt8], sharedSecret: [UInt8]) -> Data? {
+        let dataLength = data.count
+        let blockSize = kCCBlockSizeAES128
+        let len = Int(dataLength) + blockSize
+        var decryptedData = [UInt8](repeating: 0, count: len)
+
+        let keyLength = size_t(kCCKeySizeAES256)
+        if sharedSecret.count != keyLength {
+            assert(false, "unexpected shared_sec len: \(sharedSecret.count) != 32")
+            return nil
+        }
+
+        let algorithm: CCAlgorithm = UInt32(kCCAlgorithmAES128)
+        let options: CCOptions   = UInt32(kCCOptionPKCS7Padding)
+
+        var numberOfBytesDecrypted: size_t = 0
+
+        let status = CCCrypt(operation,
+                             algorithm,
+                             options,
+                             sharedSecret,
+                             keyLength,
+                             iv,
+                             data,
+                             dataLength,
+                             &decryptedData,
+                             len,
+                             &numberOfBytesDecrypted
+        )
+
+        if UInt32(status) != UInt32(kCCSuccess) {
+            return nil
+        }
+
+        return Data(bytes: decryptedData, count: numberOfBytesDecrypted)
+    }
+
+    private func randomBytes(count: Int) -> Data {
```

Comment:
We already have this function in Data+Additions, so you can remove this one and call the existing one.